services-custom/dynamodb-enhanced/src/test/java/software/amazon/awssdk/enhanced/dynamodb/mapper/testbeans/NestedImmutable.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.enhanced.dynamodb.mapper.testbeans;
+
+import java.util.Objects;
+import software.amazon.awssdk.enhanced.dynamodb.mapper.annotations.DynamoDbImmutable;
+import software.amazon.awssdk.enhanced.dynamodb.mapper.annotations.DynamoDbPartitionKey;
+import software.amazon.awssdk.enhanced.dynamodb.mapper.annotations.DynamoDbPreserveEmptyBean;
+
+@DynamoDbImmutable(builder = NestedImmutable.Builder.class)
+public class NestedImmutable {
+    private final String id;
+    private final Integer integerAttribute;
+    private final AbstractImmutable innerBean;
+
+    private NestedImmutable(Builder b) {
+        this.id = b.id;
+        this.integerAttribute = b.integerAttribute;
+        this.innerBean = b.innerBean;
+    }
+
+    @DynamoDbPartitionKey
+    public String id() {
+        return this.id;
+    }
+
+    public Integer integerAttribute() {
+        return integerAttribute;
+    }
+
+    @DynamoDbPreserveEmptyBean
```

Comment:
Technically this is not a bean. Since we're making this name up I think we should pick a more accurate description for what this thing is. 'PreserveEmptyObject'? or something like that

Reply:
+1